velox/exec/TopNRowNumber.h

```diff
@@ -220,29 +216,33 @@ class TopNRowNumber : public Operator {
   // Maximum number of rows in the output batch.
   vector_size_t outputBatchSize_;
 
+  // The below variables are used when outputting from memory.
+  // Vector of pointers to individual rows in the RowContainer for the current
+  // output block.
   std::vector<char*> outputRows_;
-
   // Number of partitions to fetch from a HashTable in a single listAllRows
   // call.
   static const size_t kPartitionBatchSize = 100;
 
   BaseHashTable::RowsIterator partitionIt_;
-
   std::vector<char*> partitions_{kPartitionBatchSize};
-
   size_t numPartitions_{0};
 
-  std::optional<int32_t> currentPartition_;
-
-  vector_size_t remainingRowsInPartition_{0};
+  // This is the index of the current partition within partitions_ which is
+  // obtained from the HashTable iterator.
+  std::optional<int32_t> currentPartitionNumber_;
+  // This is the currentPartition being output. It is possible that the
+  // partition is output across multiple output blocks.
+  TopNRowNumber::TopRows* outputPartition_{nullptr};
```

Comment:
Isn't the previous `currentPartition_` naming apt? Why change to `outputPartition_`?

Reply:
I changed it to avoid confusion with currentParitionNumber_ or the older currentPartition_. It is also only used for the output loop so outputPartition_ seemed reasonable as well.